### PR TITLE
Add support for multiple integers value for attribute and `NonNull`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
For now this implementation doesn't work. Need to figure out what I did wrong.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
gcc:

```
123r:[irange] _Bool [0, 0] NONZERO 0x0
true_range:[irange] _Bool [1, 1]
false_range:[irange] _Bool [0, 0] NONZERO 0x0
```

libgccjit:

```
123r:[irange] bool VARYING
true_range:[irange] bool [1, 1]
false_range:[irange] bool [0, 0] NONZERO 0x0
```

With this C code:

**Comment by GuillaumeGomez** [comment]
All fixed. Now remains to move the check earlier (in `add_attribute` methods directly) and to handle `no_add_attrs`.

File: gcc/jit/libgccjit.cc [hunk 1]
```diff
@@ -4234,6 +4234,18 @@ gcc_jit_function_add_string_attribute (gcc_jit_function *func, gcc_jit_fn_attrib
   func->add_string_attribute (attribute, value);
 }
 
+/* This function adds an attribute with multiple integer values. For example
+   `nonnull(1, 2)`. The numbers in `values` are supposed to map how they
+   should be written in C code. So for `nonnull(1, 2)`, you should pass `1`
+   and `2` in `values` (and set `length` to `2`). */
+void
+gcc_jit_function_add_integer_array_attribute (gcc_jit_function *func, gcc_jit_fn_attribute attribute, const int* values, int length)
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply gcc/jit/libgccjit.cc:4242]: It's always `size_t` as you said. Updating!

File: gcc/jit/jit-playback.h [hunk 2]
```diff
@@ -113,7 +115,8 @@ class context : public log_user
 		int is_variadic,
 		enum built_in_function builtin_id,
 		const std::vector<gcc_jit_fn_attribute> &attributes,
-		const std::vector<std::pair<gcc_jit_fn_attribute, std::string>> &string_attributes,
+    const std::vector<std::pair<gcc_jit_fn_attribute, std::string>> &string_attributes,
```
Comment (antoyo) [inline gcc/jit/jit-playback.h:118]: Seems like the tabs got lost here.

**Comment by GuillaumeGomez** [comment]
Questions to ask David when we'll send the attributes patch upstream:

 * Should we check attributes in the API?
 * Is it valid to use warning/error in `handle_attribute_*` functions?
   * If yes, should we add warning/error in already existing `handle_attribute_*` functions?

**Pull request merged**